lib/puppet/network/http/factory.rb

```diff
@@ -25,17 +25,7 @@ def initialize
   def create_connection(site)
     Puppet.debug("Creating new connection for #{site}")
 
-    args = [site.host, site.port]
-
-    unless Puppet::Util::HttpProxy.no_proxy?(site.addr)
-      if Puppet[:http_proxy_host] == "none"
-        args << nil << nil
-      else
-        args << Puppet[:http_proxy_host] << Puppet[:http_proxy_port]
-      end
-    end
-
-    http = Net::HTTP.new(*args)
+    http = Puppet::Util::HttpProxy.proxy(URI(site.addr))
```

Comment:
so what changed that we're now able to use the `Puppet::Util::HttpProxy.proxy` method? One of the previous PRs from this set? That part isn't super clear to me, since it looks like it's not happening in this commit, it must have happened at some other time?

Reply:
Ah `Puppet::Util::HttpProxy.proxy(URI)` does everything we need to create a `Net::HTTP` object with the correct configuration, http proxy host, port, user and password, while taking `no_proxy?` into account. So the fix is to just call that existing method, and delete the old code that didn't take proxy user/password into account